src/main/java/core/basesyntax/service/impl/ReturnOperationHandler.java

```diff
@@ -0,0 +1,22 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.Operation;
+import core.basesyntax.service.OperationHandler;
+import core.basesyntax.service.ShopDao;
+import java.util.Map;
+
+public class ReturnOperationHandler extends OperationHandler {
+    public ReturnOperationHandler(ShopDao dao) {
+        super(dao);
+    }
+
+    @Override
+    public void process(Operation operation) {
+        if (operation.getAmount() < 0) {
+            throw new RuntimeException("Illegal amount of fruits");
+        }
+        Map.Entry<String, Integer> entry = dao.get(operation.getFruitName());
+        int newAmount = entry.getValue() + operation.getAmount();
+        dao.update(operation.getFruitName(), newAmount);
```

Comment:
What will happen if someone returns fruit which does not exist in storage?
(As for me. People can return fruits from the previous day)

Reply:
In DB initializer we get all fruit types found in the file